tensorflow/go/graph_test.go

```diff
@@ -35,6 +35,28 @@ func hasOperations(g *Graph, ops ...string) error {
 	return fmt.Errorf("Graph does not have the operations %v", missing)
 }
 
+func hasOperationsList(g *Graph, ops ...string) error {
```

Comment:
Might I suggest folding this into `hasOperations`?

So `hasOperations` would do what it's doing before this change, and then in the end will do:

```go
if len(missing) != 0 {
  return fmt.Errorf("Graph does not have the operations %v", missing)
}

var map[string]bool inList;
for _, op := range g.Operations() {
  inList[op.Name()] = true;
}

for _, op := range ops {
  if !inList[op] {
    missing = append(missing, op)
  }
}

if len(missing) != 0 {
  return fmt.Errorf("Operations %v are missing from graph.Operations()", missing)
}

return nil
```

I feel a few benefits of this are:
(a) The ordering doesn't matter: In the PR right now, the list of ops provided to `hasOperationsList` has to match the order returned by `TF_GraphNextOperation`. There is no ordering guarantee there.

(b) `hasOperations` then more comprehensively tests that the graph has the operations provided.

Reply:
yes I will make this change. I didn't realize the next operation doesn't necessarily preserve ordering.